modules/apps/foundation/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/util/comparator/OrganizationMemberNameComparator.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.users.admin.web.util.comparator;
+
+import com.liferay.portal.kernel.model.Organization;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.util.OrderByComparator;
+
+/**
+ * @author Pei-Jung Lan
+ */
+public class OrganizationMemberNameComparator
+	extends OrderByComparator<Object> {
+
+	public static final String ORDER_BY_ASC = "name ASC";
+
+	public static final String ORDER_BY_DESC = "name DESC";
+
+	public static final String[] ORDER_BY_FIELDS = {"name"};
+
+	public OrganizationMemberNameComparator() {
+		this(false);
+	}
+
+	public OrganizationMemberNameComparator(boolean ascending) {
+		_ascending = ascending;
+	}
+
+	@Override
+	public int compare(Object object1, Object object2) {
+		String name1;
+		String name2;
+
+		if (object1 instanceof Organization) {
```

Comment:
I dont quite understand what you mean by 

> By having this interface they don't have to return List from their getFoldersAndDocuments() methods, but rather List.

Can you explain this more?

It'd be nice to have a common interface for users and organizations. We can simplify a lot of the checking for objects if we have a common interface. However, my concern is that if we were to add a new interface, we need to update the portal-kernel model package. We'll need to increase the version of portal.kernel.model and manually update ALL provider-type modules that uses this package to use default for portal.kernel since providers have a narrower import range of [MAJOR.MINOR.., MAJOR.MINOR+1). Unless it's absolutely necessary, I'd prefer not making changes to portal-kernel model since a lot of modules depend on it.

I think what we can do to improve this is to follow [RepositoryModelTitleComparator](https://github.com/liferay/liferay-portal/blob/master/portal-kernel/src/com/liferay/document/library/kernel/util/comparator/RepositoryModelTitleComparator.java) and have a separate method to get the name of the object. That way, we can eliminate duplication of the if clause.